include/pmacc/cuSTL/algorithm/kernel/run-time/Foreach.hpp

```diff
@@ -142,18 +143,22 @@ math::Size_t<DIM3> getBestCudaBlockDim(const math::Size_t<dim> gridDimension)
             this->_blockDim.y(),                                                                                    \
             this->_blockDim.z()                                                                                     \
         );                                                                                                          \
+        uint32_t numWorkers = traits::GetNumWorkers< cudaSpecs::MaxNumThreadsPerBlockDim::x::value >::value;        \
```

Comment:
The max value is 1024 for the x dimension and this value fits also for cpu. The CPU OpenMP accelerator which is running threads in a block in parallel will be slow down the cpu extreme but since we are not using this accelerator it is fine.

Reply:
Ah right, I did misread the naming! It's of course a block's max dim...